deps/rabbit/src/rabbit_core_ff.erl

```diff
@@ -63,25 +65,27 @@
       depends_on    => [stream_queue]
      }}).
 
+-rabbit_feature_flag(
+    {feature_flags_v2,
+     #{desc          => "Feature flags subsystem V2",
+       stability     => stable
+     }}).
+
 -rabbit_feature_flag(
    {direct_exchange_routing_v2,
     #{desc          => "v2 direct exchange routing implementation",
       stability     => stable,
-      depends_on    => [implicit_default_bindings],
+      depends_on    => [implicit_default_bindings, feature_flags_v2],
       migration_fun => {?MODULE, direct_exchange_routing_v2_migration}
      }}).
 
--rabbit_feature_flag(
-    {feature_flags_v2,
-     #{desc          => "Feature flags subsystem V2",
-       stability     => stable
-     }}).
 
 %% -------------------------------------------------------------------
 %% Direct exchange routing v2.
 %% -------------------------------------------------------------------
 
-direct_exchange_routing_v2_migration(FeatureName, _FeatureProps, enable) ->
+direct_exchange_routing_v2_migration(#ffcommand{name = FeatureName,
+                                                command = enable}) ->
```

Comment:
The function should not return the result of `rabbit_binding:populate_index_route_table()` directly because if this one changes, it will break the migration function. A better way would be to handle the return value of that call and return `ok` if it was successful, or `{error, Reason}` if it was not.